### PR TITLE
Meet Vora: Survey Visualizer Application

### PR DESCRIPTION
**Pull request by meet-vora**
Checklist for prospect students for Google Summer of Code 2016:

- [x] I contacted NumFOCUS mentors before writing my proposal
- [x] I showed you my contribution (it can be any form: proof-of-concept project idea, some sample code or just a link to your commits from other project)
- [x] I linked to my sample contribution from the proposal
- [x] I linked to my opened issues in numfocus/gsoc repository from the proposal

### CODE REVIEW

File: 2016/proposals/vora-meet.md [hunk 1]
```diff
@@ -0,0 +1,162 @@
...
+
+#### **Database**: PostgreSQL
+With the release of v9.4, extensive support for JSON has been added to PostgreSQl, making it the perfect solution for our idea. PostgreSQL can easily handle large amounts of data and thus, speed performance and fetching of newer data wouldn't be an issue.  
+Further, as per the discussions with the mentors, I have been made aware that SQLite is currently being employed in AMY. Thus, we can currently store the JSON as a string in PostgreSQL, to maintain cross-database compatiblity.  
+To enable support for simple queries, we will save certain data (e.g. *date of workshop*) in seperate fields in the models.
+
+#### **Frontend**: Bootstrap + Highcharts / Plotly.js
```
Comment (rgaiacs) [inline 2016/proposals/vora-meet.md]: Highcharts is open source but not free.

**Comment by meet-vora** [comment]
Oops. Am sorry for missing out the point that Software Carpentry isn't *Non-Commercial*

>Do you want to use products for a personal website, a school site or a non-profit organisation? Then you don't need the author's permission, just go on and download.

I have made the necessary rectification.
I request you and @pbanaszkiewicz to kindly review the proposal and provide your feedback.
Thanking You.

Regards,
Meet Vora

**Comment by pbanaszkiewicz** [comment]
> Oops. Am sorry for missing out the point that Software Carpentry isn't *Non-Commercial*

I'm puzzled. Software Carpentry Foundation *is* non-profit as far as I know…

@meet-vora: I think Raniere didn't know about non-commercial use of Highcharts, so you can include them; if you do, you can include pros/cons of using Highcharts compared to plotly.js.

File: 2016/proposals/vora-meet.md [hunk 2]
```diff
@@ -0,0 +1,161 @@
...
+
+#### **Database**: PostgreSQL
+With the release of v9.4, extensive support for JSON has been added to PostgreSQl, making it the perfect solution for our idea. PostgreSQL can easily handle large amounts of data and thus, speed performance and fetching of newer data wouldn't be an issue.  
+Further, as per the discussions with the mentors, I have been made aware that SQLite is currently being employed in AMY. Thus, we can currently store the JSON as a string in PostgreSQL, to maintain cross-database compatiblity.  
+To enable support for simple queries, we will save certain data (e.g. *date of workshop*) in seperate fields in the models.
```
Comment (pbanaszkiewicz) [inline 2016/proposals/vora-meet.md]: Okay, my guts tell me the most crucial part of the proposal lies in the good database schema.

Please:

1. expand your SQLite / cross-db compatible design
2. expand your PostgreSQL-only design

You can argue which one you prefer and why.

Thanks!

File: 2016/proposals/vora-meet.md [hunk 3]
```diff
@@ -0,0 +1,161 @@
...
+This time will be used to perform complete *requirement analysis* of the project and  to better know the Software Carpentry community - the users as well as the developers. During this period, the discussions will be held with mentors, as part of the analysis phase, to finalize the minute details including:
+* presentation layout for each question-type and answer-type
+* choice of plotting library
+* database schema
+* way to present the choices (of 5 surveys) when the workshop ID is provided 
+
+
+### May 25th -  June 7th
+*Begin coding.*  
+As I have implemented OAuth during the discussion period and have good experience of coding in Django, I will be developing the backend and use SurveyMonkey OAuth to authenticate the application with Software Carpentry's SurveyMonkey account.
```
Comment (pbanaszkiewicz) [inline 2016/proposals/vora-meet.md]: I don't think you need to spend 2 weeks on authentication.

**Comment by pbanaszkiewicz** [comment]
@meet-vora:

Please provide a timezone you should be available from during the summer and make a note when your availability (and to what extent) starts (ie. when you finish your school term, how many hours per week can you spend on the project, etc.).

**Comment by rgaiacs** [comment]
> I think Raniere didn't know about non-commercial use of Highcharts, so you can include them; if you do, you can include pros/cons of using Highcharts compared to plotly.js.

I knew. MIT and BSD licenses and GPL are easy to understand. Licenses that only allow free use by non-commercial projects create a grey area that i prefer to not enter without a good lawyer.

**Comment by pbanaszkiewicz** [comment]
> I knew. MIT and BSD licenses and GPL are easy to understand. Licenses that only allow free use by non-commercial projects create a grey area that i prefer to not enter without a good lawyer.

In that case I'm -1 on Highcharts as well…

**Comment by meet-vora** [comment]
Hi @pbanaszkiewicz and @rgaiacs.

I've made the necessary additions. I request you to kindly provide your feedback.
Thanking You.

Regards,
Meet Vora

File: 2016/proposals/vora-meet.md [hunk 4]
```diff
@@ -0,0 +1,209 @@
...
+ Question Type   |      Possible way of presenting the answers     
+----------|-------------
+ Single choice (*Radio* / *Menu*) |  Bar graph / Pie chart 
+ Multiple choice |  Bar graph / Pie chart 
+ Matrix Menu | Multiple Bar graphs / Stacked Bar graph / Grouped Bar chart 
+ Matrix table | A Bar graph for each record / Matrix "chart
+ Text field | Text content (presented in required format)
+ Rating-type Question | Bar graph / Pie chart 
+
+In later stages, the application will be integrated with [AMY](https://github.com/swcarpentry/amy), a django-based workshop administration application for Software Carpentry.
```
Comment (rgaiacs) [inline 2016/proposals/vora-meet.md]: Django, with capital D.

File: 2016/proposals/vora-meet.md [hunk 5]
```diff
@@ -0,0 +1,209 @@
...
+However, as per the discussions with the mentors, I have been made aware that SQLite is currently being employed in AMY. With the discussed database schema, we can store the JSON as a string in SQLite.  
+
+#### **Frontend**: Bootstrap + Plotly.js
+The reason for narrowing down the choices out of various charting libraries to a single one is the fact that this library enables developers to design responsive, customizable charts and adds functionalities to the already large feature set of *D3.js*, on which it is based. Further, (i) it is completely free & open-source (ii) *Plotly playground*: developers can run write the code online & modify using GUI (thus, modify, enhance & test charts well before using them) *and* (iii) it is easier to use & presence of few rendering bugs in Highcharts [1](https://news.ycombinator.com/item?id=10583784) [2](http://stackoverflow.com/questions/6924356/highcharts-jquery-rendering-problem-all-browsers)
```
Comment (rgaiacs) [inline 2016/proposals/vora-meet.md]: "in Highcharts" -> "than Highcharts"

File: 2016/proposals/vora-meet.md [hunk 6]
```diff
@@ -0,0 +1,209 @@
...
+
+## Other Details
+#### Summer Schedule
+I have no other commitments scheduled for the summer and would devote atleast 50 hours a week, and more if required.
```
Comment (rgaiacs) [inline 2016/proposals/vora-meet.md]: - "atleast" -> "at least"
- You just need to devote 40 hours a week.

**Comment by rgaiacs** [comment]
> I request you to kindly provide your feedback.

@meet-vora Looks good. I commented a few typos that you can fix. Please don't forgot to submit your proposal on http://summerofcode.withgoogle.com/. We can't do anything if your proposal isn't on Googles database.

File: 2016/proposals/vora-meet.md (hunk 4 above)
Comment (pbanaszkiewicz) [inline 2016/proposals/vora-meet.md]: https://www.youtube.com/watch?v=-Ae590hensE

File: 2016/proposals/vora-meet.md (hunk 6 above)
Comment (pbanaszkiewicz) [inline 2016/proposals/vora-meet.md]: I don't know if GSOC rules set a certain number of hours/week expected – but 50 hours/week is too much (unhealthy). From my experience as a GSOC student, it'd be more like 30+ hours (sometimes students are blocked by mentors).

Comment (rgaiacs) [inline 2016/proposals/vora-meet.md]: > I don't know if GSOC rules set a certain number of hours/week expected

We have that [on the FAQ](https://developers.google.com/open-source/gsoc/faq#how_much_time_does_gsoc_participation_take):

> How much time does GSoC participation take?
>
> You are expected to spend about 40 hours a week working on your project during the 3 month coding period. If you already have an internship, another summer job, or plan to be gone on vacation for more than a week during that time, GSoC is not the right program for you this year.

**Comment by meet-vora** [comment]
So, should I directly upload the PDF of the markdown file, skipping the draft?

**Comment by pbanaszkiewicz** [comment]
@meet-vora if you feel this proposal is complete, you can go straight to uploading the final PDF.

**Comment by meet-vora** [comment]
Thank you @pbanaszkiewicz and @rgaiacs for your time and help.

Regards,
Meet Vora

**Comment by pbanaszkiewicz** [comment]
@meet-vora you're welcome. And thank you for applying to us :-)

**Comment by meet-vora** [comment]
@pbanaszkiewicz Just a small doubt here. 
How is the PDF supposed to be named?
Is it 'Meet Vora: Survey Visualizer Application'?

**Comment by pbanaszkiewicz** [comment]
@meet-vora it probably doesn't matter since I can see it's coming from your account on GSOC website.

File: 2016/proposals/vora-meet.md [hunk 7]
```diff
@@ -105,22 +105,23 @@ class WorkshopSurvey(models.Model):
...
+		{
+			'id': 		question_id,
+			'answer':	[
+							{
+								'id': 		answer_id
+								'count':	answer_count['answer_id']	 	 #answer_count shall be user-defined dictionary as described above
+							},
+						]
+		},
+}
```
Comment (pbanaszkiewicz) [inline 2016/proposals/vora-meet.md]: Minor thing: it's better to use space-based indentation (with the indent of 4 spaces).

File: 2016/proposals/vora-meet.md [hunk 8]
```diff
@@ -190,20 +190,20 @@ The documentation will be finalized. Code will be beautified along with removal
...
-I have no other commitments scheduled for the summer and would devote atleast 50 hours a week, and more if required.
-The ongoing semester will be completed by 10th of May. The next semester is scheduled to begin on 20th of July, 2016. 
+I have no other commitments scheduled for the summer and would devote at least 40 hours a week, and more if required.
+The ongoing semester will be completed by 6th of May. The next semester is scheduled to begin on 20th of July, 2016. 
 
 #### Contact
 Email: [meetvora@outlook.com](mailto:meetvora@outlook.com) / [v.h.c.meetirth@gmail.com](mailto:v.h.c.meetirth@gmail.com)  
-Mobile: +91 839 487 7088
\ No newline at end of file
+Mobile: +91 839 487 7088 / +91 702 198 4036
```
Comment (pbanaszkiewicz) [inline 2016/proposals/vora-meet.md]: I didn't spot this before, but you aren't required to provide your phone number in a public place like this proposal.

Comment (pbanaszkiewicz) [inline 2016/proposals/vora-meet.md]: Uh, maybe it's my fault, but somehow this ↑ comment was placed under wrong line…

**Comment by pbanaszkiewicz** [comment]
Thanks for the update, @meet-vora.

I'm checking with you to make sure that you submitted your final proposal to the GSOC website, as the deadline is soon.

Thanks again and cheers!

File: 2016/proposals/vora-meet.md (hunk 7 above)
Comment (meet-vora) [inline 2016/proposals/vora-meet.md]: I have incorporated the changes into the final PDF.

**Comment by meet-vora** [comment]
Hi @pbanaszkiewicz,
I have incorporated the changes into the **.md** file and have (re-)submitted the PDF.
I request you to kindly check the PDF.
Thank you for your continuous feedback and time. It was a pleasure interacting with you. 

Regards,
Meet Vora

**Comment by pbanaszkiewicz** [comment]
@meet-vora:
>  I have incorporated the changes into the .md file and have (re-)submitted the PDF.
> …
>  Thank you for your continuous feedback and time. It was a pleasure interacting with you. 

Thanks a lot!

>  I request you to kindly check the PDF.

We (mentors) can't see the final proposal until after the deadline.

**Comment by rgaiacs** [comment]
@meet-vora Thanks.

**Pull request merged**